warehouse/ingest-scripts/src/main/resources/bin/ingest/load-job-cache.sh

```diff
@@ -42,9 +42,23 @@ sed s%${BEFORE}%${AFTER}% job-cache-env.sh > job-cache-env.tmp
 
 date
 
+function no_ctrlc() {
+  let count++
+  echo
+  if [[ $count == 1 ]]; then
+    echo "WARNING: Killing this script will leave the job cache in a most-likely unusable state. You will need to rerun this script.
+    To continue stopping, please issue another SIGTERM."
+  else
+    echo "Second SIGTERM received, exiting script."
+    rm -r -f "$tmpdir"
+    exit 1
+  fi
+}
+
 # prepare a directory with links to all of the files/directories to put into the jobcache
 tmpdir=`mktemp -d $MKTEMP_OPTS`
-trap 'rm -r -f "$tmpdir"; exit $?' INT TERM EXIT
+trap 'rm -r -f "$tmpdir"; exit 1' INT EXIT
```

Comment:
Doesn't 
```
trap 'blah ; exit 1' EXIT
```
guarantee that the script will _always_ exit w/ non-zero status?

Reply:
going off a discussion with Jeff it looks like if TERM/KILL signals are being sent while this script is running it probably makes sense to return with a non-zero status to imply the script did not exit clean.

Reply:
Trap arg for `EXIT` will fire even on normal exit of the script. So, I don't see how we'll ever get a zero exit status

E.g., if you run the script below, you'll get non-zero return status:
```
#!/bin/bash
trap 'echo trapping; exit 1' INT EXIT
exit 0
```

Reply:
Ah ok, now I follow. I'll see if I can extract out the EXIT signal from that logic